fastapi_websocket_rpc/websocket_rpc_client.py

```diff
@@ -15,6 +15,149 @@
+    """
+    def __init__(self):
+        self._websocket = None
+
+    """
+    Args:
+        **kwargs: Additional args passed to connect
+            https://websocket-client.readthedocs.io/en/latest/examples.html#connection-options
+            https://websocket-client.readthedocs.io/en/latest/core.html#websocket._core.WebSocket.connect
+    """
+    async def connect(self, uri: str, **connect_kwargs):
+        self._websocket = await asyncio.get_event_loop().run_in_executor(None, websocket.create_connection, uri, **connect_kwargs)
+
+    async def send(self, msg):
+        if self._websocket is None:
+            # connect must be called before.
+            logging.error("Websocket connect() must be called before.")
+        await asyncio.get_event_loop().run_in_executor(None, self._websocket.send, msg)
+
+    async def recv(self):
+        if self._websocket is None:
+            # connect must be called before.
+            logging.error("Websocket connect() must be called before.")
+        try:
+            msg = await asyncio.get_event_loop().run_in_executor(None, self._websocket.recv)
+        except websocket.WebSocketConnectionClosedException as err:
+            # websocket.WebSocketConnectionClosedException means remote host closed the connection or some network error happened
+            raise BaseConnectionClosedException from err
+        return msg
+
+    async def close(self, code: int = 1000):
+        if self._websocket is not None:
+            # Case opened, we have something to close.
+            self._websocket.close(code)
+
+    async def handle_exception(self, exception: Exception):
+        try:
+            raise exception
+        # See https://websocket-client.readthedocs.io/en/latest/exceptions.html
+        except websocket._exceptions.WebSocketAddressException:
+            logger.info("websocket address info cannot be found")
+            raise
+        except websocket._exceptions.WebSocketBadStatusException:
+            logger.info("bad handshake status code")
+            raise
+        except websocket._exceptions.WebSocketConnectionClosedException:
+            logger.info("remote host closed the connection or some network error happened")
+            raise
+        except websocket._exceptions.WebSocketPayloadException:
+            logger.info(
+                f"WebSocket payload is invalid")
+            raise
+        except websocket._exceptions.WebSocketProtocolException:
+            logger.info(f"WebSocket protocol is invalid")
+            raise
+        except websocket._exceptions.WebSocketProxyException:
+            logger.info(f"proxy error occurred")
+            raise
+        except OSError as err:
+            logger.info("RPC Connection failed - %s", err)
+            raise
+        except Exception as err:
+            logger.exception("RPC Error")
+            raise
+
+class WebSocketsClientHandler(SimpleWebSocket):
```

Comment:
Maybe we should give these a better name  - the difference of just one letter "S" can be confusing.
Maybe `ProxyEnabledWebSocketClientHandler` ?

Reply:
I thought there might be other reasons for users to switch to websocket-client, other than proxy. I don't know websocket-client enough. But ok to rename it. Done!

Reply:
https://github.com/antoinetran/fastapi_websocket_rpc/commit/385df4392aca04a49ff934ed349f8f3eb1b30495

Reply:
Cool - @antoinetran :) 

@roekatz can you give this an additional look - this being sensitive code - it could use another eye